dango/types/src/lending.rs

```diff
@@ -10,6 +10,46 @@ pub static NAMESPACE: LazyLock<Part> = LazyLock::new(|| Part::new_unchecked("len
 /// Sub-namespace that liquidity share tokens will be minted under.
 pub static SUBNAMESPACE: LazyLock<Part> = LazyLock::new(|| Part::new_unchecked("pool"));
 
+// -------------------------------- Bounds -------------------------------------
+
+/// Defines the bounds for a loan-to-value ratio: 0 < LoanToValue < 1.
+#[grug::derive(Serde)]
+pub struct LoanToValueBounds;
+
+impl Bounds<Udec128> for LoanToValueBounds {
+    const MAX: Option<Bound<Udec128>> = Some(Bound::Exclusive(Udec128::ONE));
+    const MIN: Option<Bound<Udec128>> = Some(Bound::Exclusive(Udec128::ZERO));
+}
+
+/// A decimal bounded by the loan-to-value bounds.
+pub type LoanToValue = Bounded<Udec128, LoanToValueBounds>;
+
+/// Defines the bounds for a collateral power: 0 < CollateralPower <= 1.
+#[grug::derive(Serde)]
+pub struct CollateralPowerBounds;
+
+impl Bounds<Udec128> for CollateralPowerBounds {
+    const MAX: Option<Bound<Udec128>> = Some(Bound::Inclusive(Udec128::ONE));
+    const MIN: Option<Bound<Udec128>> = Some(Bound::Exclusive(Udec128::ZERO));
+}
+
+/// A decimal bounded by the collateral power bounds.
+pub type CollateralPower = Bounded<Udec128, CollateralPowerBounds>;
```

Comment:
As commented earlier, collateral power is not the concern of the lending contract, so it shouldn't be in this file. I'm going to move this to the `account/margin.rs`.